test/e2e_node/container_log_rotation_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	kubelogs "k8s.io/kubernetes/pkg/kubelet/logs"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	testContainerLogMaxFiles    = 3
+	testContainerLogMaxSize     = "40Ki"
+	rotationPollInterval        = 5 * time.Second
+	rotationEventuallyTimeout   = 3 * time.Minute
+	rotationConsistentlyTimeout = 2 * time.Minute
+)
+
+var _ = framework.KubeDescribe("ContainerLogRotation [Slow] [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("container-log-rotation-test")
+	Context("when a container generates a lot of log", func() {
+		BeforeEach(func() {
+			if framework.TestContext.ContainerRuntime != kubetypes.RemoteContainerRuntime {
+				framework.Skipf("Skipping ContainerLogRotation test since the container runtime is not remote")
+			}
+		})
+
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FeatureGates[string(features.CRIContainerLogRotation)] = true
```

Comment:
even though you set this here, I think you may also need to set it in the test-infra config for the containerd serial suite.

Reply:
It seems to be working for my local environment. :)
@mtaufen Do we need to do this?

Reply:
Do you expect log rotation to be on, in general, for the serial suite, or just for this test case? 
If you want it on in general, you need to set it at a higher level (preferably in the test-infra config). 
If just for this test case, here should be fine as long as toggling the gate doesn't require some special handling, like a node restart.